### PR TITLE
Convert zperf to use socket services API

### PR DESCRIPTION
**Pull request by jukkar**
I converted the zperf library to use socket services API in order to save memory. After the conversion, in my test setup (qemu_x86) 2836 bytes (ram) were saved.

For the zperf library and sample, the savings look like this:
```
main zperf rom_report 13602 ram_report 10120
pr   zperf rom_report 14081 ram_report  6184
                      =====            =====
                       -479             3936
```
So slight increasing for ROM usage, but quite substantial savings for RAM. The socket service implementation needs also memory so the total savings are a bit smaller.

### CODE REVIEW

**Comment by jukkar** [comment]
Added a way to tweak the socket service dispatcher thread priority, this way it is possible to get best performance out of the stack.

File: subsys/net/lib/zperf/zperf_tcp_receiver.c [hunk 1]
```diff
@@ -99,6 +96,139 @@ static void tcp_received(const struct sockaddr *addr, size_t datalen)
...
+			NET_ERR("recv failed on IPv%d socket (%d)",
+				family == AF_INET ? 4 : 6,
+				errno);
+			tcp_session_error_report();
+			/* This will close the zperf session */
+			ret = 0;
+		}
+
+		for (; i < SOCK_ID_MAX; i++) {
+			if (fds[i].fd == pev->event.fd) {
```
Comment (rlubos) [inline subsys/net/lib/zperf/zperf_tcp_receiver.c:184]: Would it be possible that we won't find FD on the list? Just wonder if we should check if `i == SOCK_ID_MAX` after the loop, and possibly handle it as an error?

**Review by rlubos — COMMENTED** [review]
Looks good, just one question.

File: subsys/net/lib/zperf/zperf_tcp_receiver.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/lib/zperf/zperf_tcp_receiver.c:184]: Good point, there was a check in the listen socket branch but the check was missing here, I will add one.

**Review by rlubos — DISMISSED** [review]

**Comment by jukkar** [comment]
Fixed the CI warning

File: subsys/net/lib/sockets/Kconfig [hunk 2]
```diff
@@ -86,6 +86,27 @@ config NET_SOCKETS_SERVICE
...
 	  depending on network configuration.
 
+config NET_SOCKETS_SERVICE_THREAD_PRIO
+	int "Priority of the socket service dispatcher thread"
+	default NUM_PREEMPT_PRIORITIES
+	depends on NET_SOCKETS_SERVICE
+	help
+	  Set the priority of the socket service dispatcher thread. This handler
+	  polls the sockets and either places the triggered socket to work queue
+	  for asynchrohous handlers, or calls the user supplied callback directly
```
Comment (pdgendt) [inline subsys/net/lib/sockets/Kconfig:96]:
```suggestion
	  for asynchronous handlers, or calls the user supplied callback directly
```

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**